src/metax_api/tests/api/oaipmh/minimal_api.py

```diff
@@ -321,6 +321,19 @@ def test_get_urnresolver_record(self):
                                         self.preferred_identifier)
         self.assertTrue(len(identifiers) == 1, response.content)
 
+    def test_legacy_catalog_datasets_are_not_urnresolved(self):
+        cr = CatalogRecord.objects.get(identifier=self.identifier)
+        cr.data_catalog.catalog_json['identifier'] = settings.LEGACY_CATALOGS[0]
+        cr.data_catalog.force_save()
+
+        response = self.client.get(
```

Comment:
In practice this call should never be used by urn.fi or any other service, but for this test case it is ok to use, since we are only interested whether the cr is resolved or not.

Longer story: When using verb=ListRecords one must use set=urnresolver parameter to get THE list of the proper urn-resolvable crs. If you don't use it, you only get things related to records that have NOT been deleted. When using this verb=GetRecord, giving set-parameter is not allowed, so currently the code only serves non-deleted crs for verb=GetRecord, especially when using metadataPrefix=oai_dc_urnresolver. So basically since this verb=ListRecords&metadataPrefix=oai_dc_urnresolver&set=urnresolver is the only appropriate list of records that should be used by anyone, in future the oai-server code might be changed so that set=urnresolver is no longer needed, and by giving this metadataPrefix=oai_dc_urnresolved the crs would be searched like they are currently searched with set=urnresolved, i.e. deleted crs would be also taken into account.